docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+
+This process is similar to what happens internally when calling fit on `TabularPredictor`, but is simplified and minimal.
+
+If the data was already cleaned (all numeric), then we could call fit directly with the data, but the adult dataset is not.
+
+### Clean labels
+
+The first step to making the input data as valid input to the model is to clean the labels.
+
+Currently, they are strings, but we need to convert them to numeric values (0 and 1) for binary classification.
+
+Luckily, AutoGluon already implements logic to both detect that this is binary classification (via `infer_problem_type`), and a converter to map the labels to 0 and 1 (`LabelCleaner`):
+
+```{.python .input}
+# Separate features and labels
+X = train_data.drop(columns=[label])
+y = train_data[label]
+X_test = test_data.drop(columns=[label])
+y_test = test_data[label]
+
+from autogluon.core.data import LabelCleaner
+from autogluon.core.utils import infer_problem_type
+# Construct a LabelCleaner to neatly convert labels to float/integers during model training/inference, can also use to inverse_transform back to original.
+problem_type = infer_problem_type(y=y)  # Infer problem type (or else specify directly)
+label_cleaner = LabelCleaner.construct(problem_type=problem_type, y=y)
+y_clean = label_cleaner.transform(y)
+
+print(f'Labels cleaned: {label_cleaner.inv_map}')
+print(f'inferred problem type as: {problem_type}')
+print('Cleaned label values:')
+y_clean.head(5)
+```
+
+### Clean features
+
+Next, we need to clean the features. Currently, features like 'workclass' are object dtypes (strings), but we actually want to use them as categorical features. Most models won't accept string inputs, so we need to convert the strings to numbers.
+
+AutoGluon contains an entire module dedicated to cleaning, transforming, and generating features called [autogluon.features](../../api/autogluon.features.html). Here we will use the same feature generator used internally by `TabularPredictor` to convert the object dtypes to categorical and minimize memory usage.
+
+```{.python .input}
+from autogluon.features.generators import AutoMLPipelineFeatureGenerator
+
+feature_generator = AutoMLPipelineFeatureGenerator()
+X_clean = feature_generator.fit_transform(X)
+
+X_clean.head(5)
+```
+
+### Fit model
+
+We are now ready to fit the model with the cleaned features and labels.
+
+```{.python .input}
+custom_model = CustomRandomForestModel()
+# We could also specify hyperparameters to override defaults
+# custom_model = CustomRandomForestModel(hyperparameters={'max_depth': 10})
+custom_model.fit(X=X_clean, y=y_clean)  # Fit custom model
+
+# To save to disk and load the model, do the following:
+# load_path = custom_model.path
+# custom_model.save()
+# del custom_model
+# custom_model = CustomRandomForestModel.load(path=load_path)
+```
+
+### Predict with fit model
```

Comment:
when it's being used in past-tense, I'd replace "fit" with "previously fit" or "trained", otherwise can be confusing

Reply:
replaced with 'trained'